sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java

```diff
@@ -616,7 +626,11 @@ public Exception buildExceptionModel() {
         }, properties, attachments);
     }
 
-    private synchronized void queueException(@NonNull final ExceptionModelBuilder exceptionModelBuilder, Map<String, String> properties, final Iterable<ErrorAttachmentLog> attachments) {
+    private synchronized UUID queueException(@NonNull final ExceptionModelBuilder exceptionModelBuilder, Map<String, String> properties, final Iterable<ErrorAttachmentLog> attachments) {
+
+        /* Snapshot userId as early as possible. */
```

Comment:
Can you explain why do you want to do it as early as possible?

Reply:
Because post happens on background thread and userId could change.
We fixed something similar in 1DS long time ago.